### ISSUE TITLE
Rancher metadata returns only "host" for self

### ISSUE THREAD

**Issue by miv** [rancher/rancher#6833]
After upgrading to 1.2.0-rc1 on several hosts rancher-metadata returns only "host" part of "self" route:

```
root@69c951a65393:/# curl --header 'Accept: application/json' "http://rancher-metadata/latest/self" | jq "."
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   536  100   536    0     0   112k      0 --:--:-- --:--:-- --:--:--  130k
{
  "host": {
    "agent_ip": "118.211.24.116",
    "hostname": "Ubuntu-1610-yakkety-64-minimal",
    "labels": {
      "db-server": "true",
      "games-webs-lb": "true",
      "hostname": "db1",
      "io.rancher.host.agent_image": "rancher/agent:v1.1.0-rc16",
      "io.rancher.host.docker_version": "1.12",
      "io.rancher.host.kvm": "true",
      "io.rancher.host.linux_kernel_version": "4.8",
      "switch": "s1",
      "type": "db"
    },
    "local_storage_mb": 441734590,
    "memory": 67380445184,
    "milli_cpu": 8000,
    "name": "db1",
    "uuid": "4a8967d7-a1e7-40a2-834d-6ef6bbfdf06f"
  }
}
```

Restarting agent doesn't help :(

EDIT: Upgrading to 1.2.0-rc2 does not help

EDIT: rancher-metadata show this in logs

```
time="2016-11-30T12:19:34Z" level=info msg="Error: /self/container/name" client=172.17.0.6 version=latest
time="2016-11-30T12:19:34Z" level=info msg="Error: /self/container/primary_ip" client=172.17.0.6 version=2015-07-25
time="2016-11-30T12:19:34Z" level=info msg="Error: /self/service/name" client=172.17.0.6 version=2015-07-25
time="2016-11-30T12:19:34Z" level=info msg="Error: /self/container/name" client=172.17.0.6 version=2015-07-25
```

EDIT: After delete rancher angent, installing anew, and reboot it has been fine
---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.2.0-rc1` Cattle: `v0.174.0` UI: `v1.2.32` |
|Access|`localauth` `admin`|
|Orchestration|`Cattle`|
|Route|`stack.index`|

**Comment by vincent99** [comment]
@miv what version were you upgrading from?  I tried this from 1.1.4 and metadata didn't disappear...

**Comment by miv** [comment]
@vincent99  It was upgrade from 1.2.0-pre6. I still has another host running with same bug. 

```
curl "http://rancher-metadata/latest/self"
host/
```

---

SO! I've tried to reboot this host and it has't come up. I've updated kernel and docker - nope. Rancher-agent either could't start or it started but could't start rancher-metadata and other `rancher/` services. I decided to upgrade it - it was Ubuntu 16.04. I've recreated it with 16.10, installed docker.io with all recommended packages and everything was alright ever since.
